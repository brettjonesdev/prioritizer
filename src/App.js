import React from 'react'
import EnterApiKey from './ui/screens/EnterApiKey'
import './App.scss'
import './styles/index.scss'
import { selectApiKey } from './ui/screens/EnterApiKey/state'
import { useSelector } from 'react-redux'
import LoggedIn from './ui/screens/LoggedIn'
import AppHeader from './ui/components/AppHeader'
import { BrowserRouter as Router } from 'react-router-dom'
import { library } from '@fortawesome/fontawesome-svg-core'
import { fas } from '@fortawesome/free-solid-svg-icons'

library.add(fas)

function App() {
  const apiKey = useSelector(selectApiKey)

  return (
    <div className="App">
      <Router>
        <AppHeader />
        <div className="content">{apiKey ? <LoggedIn /> : <EnterApiKey />}</div>
      </Router>
    </div>
  )
}

export default App
