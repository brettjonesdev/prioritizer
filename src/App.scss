@import './styles/consts';

.App {
  .content {
		max-width: $maxWidth;
		min-height: calc(100vh - 125px);
		margin: 0 auto;
		padding: $spacing;
	}
}
