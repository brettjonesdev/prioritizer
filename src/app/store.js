import { configureStore } from '@reduxjs/toolkit'
import counter from '../features/counter/counterSlice'
import apiKey, { getLocal } from '../state/slices/apiKey'
import repos from '../ui/screens/LoggedIn/Repos/reposSlice'

export default configureStore({
  preloadedState: {
    apiKey: getLocal(),
  },
  reducer: {
    counter,
    apiKey,
    repos,
  },
})
