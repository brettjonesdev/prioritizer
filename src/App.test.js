import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from './app/store'
import App from './App'
import renderer from 'react-test-renderer'

test('renders EnterApiKey view by default', () => {
  const { getByText } = render(
    <Provider store={store}>
      <App />
    </Provider>
  )

  expect(getByText(/Personal Access Token/i)).toBeInTheDocument()
})

test('renders App snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
