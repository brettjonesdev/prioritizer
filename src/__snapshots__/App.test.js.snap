// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App snapshot 1`] = `
<div
  className="App"
>
  <div
    className="AppHeader"
  >
    <div
      className="title"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Prioritizer
      </a>
    </div>
  </div>
  <div
    className="content"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="EnterApiKey"
      >
        <div>
          <label
            htmlFor="apiKey"
          >
            Enter a valid GitHub Personal Access Token
          </label>
        </div>
        <div
          className="flex-row"
        >
          <input
            className="api-key"
            id="apiKey"
            onChange={[Function]}
            type="text"
            value=""
          />
          <input
            className="submit-button primary"
            disabled={true}
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
